Allow HTTP Headers in Tracer

magikid opened the pull request:
This adds support for HTTP Headers when extracting and injecting the
context.  To use, the carrier needs to implement the
`RequestInterface` interface.

Issue: #22

cjone0102 committed:
Allow HTTP Headers in Tracer

This adds support for HTTP Headers when extracting and injecting the
context.  To use, the carrier needs to implement the
`RequestInterface` interface.

Issue: #22

jcchavezs commented:
Hi @magikid thanks for pointing our this. Could you cover https://github.com/jcchavezs/zipkin-php-opentracing/pull/23/files#diff-25c66202aad94dec22819d4ca7e347b2R74 with a test?

cjone0102 committed:
Adds test for HTTP_HEADERS

This adds a test to verify that the HTTP_HEADERS are properly extracted
from a request.

Issue: #22

magikid commented:
I don't know why I didn't when I first submitted this.  All I can say was that it was Friday 😄

jcchavezs commented:
Not sure why travis is failing. Investigating it.

magikid commented:
I see why. Style violations from my code. I'll update the PR in a few hours.

jcchavezs commented:
@magikid have you been able to sort this out?

cjone0102 committed:
Merge branch 'master' of github.com:jcchavezs/zipkin-php-opentracing

cjone0102 committed:
Fixed style violations

magikid commented:
Well it took me significantly longer than a few hours to resolve those style issues but the checks are all passing now.

jcchavezs commented:
Great work @magikid, do you have any input @ellisv (or @eligijusvitkauskas-home24)?

jcchavezs commented on tests/Unit/HeaderClass.php:
```diff
@@ -0,0 +1,89 @@
+use Psr\Http\Message\StreamInterface;
+use Psr\Http\Message\UriInterface;
+
+class HeaderClass implements RequestInterface
```
nitpick, do you mind making this class final?

jcchavezs commented on tests/Unit/HeaderClass.php:
```diff
@@ -0,0 +1,89 @@
+
+class HeaderClass implements RequestInterface
+{
+    /** @var array $headers */
```
Could we please do 
```
/**
 *@var array $headers
 */
```

instead?

jcchavezs commented on src/ZipkinOpenTracing/Tracer.php:
```diff
@@ -70,6 +71,11 @@ public function startActiveSpan($operationName, $options = [])
             $options = StartSpanOptions::create($options);
         }
 
+        if (!$this->hasParentInOptions($options) && $this->getActiveSpan() !== null) {
```
Could we please reflect this piece of logic in the unit test?

cjone0102 committed:
Makes test HeaderClass final

cjone0102 committed:
Fix style violations

jcchavezs commented:
Thanks for this @magikid, I left some comments, once they are fixed I am happy to merge this.

cjone0102 committed:
Adds intellij to gitignore

jcchavezs commented:
Merged with #29. Thanks a lot @magikid!
